flowmanager/models.py

```diff
@@ -266,24 +273,27 @@ def list_pipelines(self):
 
     def check_flow_status(self, flow_id):
         with self.session_scope() as session:
-            failed = session.query(Pipelines).filter_by(
-                flow_id=flow_id, status=STATE_FAILED).first()
-            if failed is not None:
-                return STATE_FAILED
-
             running = session.query(Pipelines).filter_by(
                 flow_id=flow_id, status=STATE_RUNNING).first()
+            # at least one running -> Flow in Progress
             if running is not None:
                 return STATE_RUNNING
-
             success = session.query(Pipelines).filter_by(
                 flow_id=flow_id, status=STATE_SUCCESS).first()
             pending = session.query(Pipelines).filter_by(
                 flow_id=flow_id, status=STATE_PENDING).first()
+            # at least one pending and success -> Flow in Progress
             if (pending is not None) and (success is not None):
                 return STATE_RUNNING
-
-            # If non of success, failed or running status is queued/pending
+            failed = session.query(Pipelines).filter_by(
```

Comment:
we never return `STATE_SUCCESS` here?

Reply:
(probably means there's also a missing test, no?)

Reply:
We return it at the end (if none of the others is satisfied). We do have tests for it